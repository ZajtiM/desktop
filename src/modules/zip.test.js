const fs = require('fs')
const path= require('path');
const {ZipData} = require ('./zip');

describe( 'unzip' ,()=> {

    test ( 'positive' ,async () => {
        let data = fs.readFileSync('testdata/success.zip');
        let zip = new ZipData(data);
        let status = await zip.extract(path.join('testdata','./out'));
        expect(status.success).toBe(true);
        expect(status.files.sort()).toMatchSnapshot();

    })

    test ( 'can be called twice' ,async () => {
        let data = fs.readFileSync('testdata/success.zip');
        let zip = new ZipData(data);
        let status1 = await zip.extract(path.join('testdata','./out'));
        expect(status1.success).toBe(true);

        zip = new ZipData(data);
        let status2 = await zip.extract(path.join('testdata','./out'));
        expect(status2.success).toBe(true);
        expect(status2.files.sort()).toEqual(status1.files.sort())

    })

    test ( 'failure' ,async () => {
        let data = fs.readFileSync('testdata/failure.zip',{encoding:null});
        let zip = new ZipData(data);
        let status = await zip.extract(path.join('testdata','./out'));
        expect(status.success).toBe(false);
        
    })

})