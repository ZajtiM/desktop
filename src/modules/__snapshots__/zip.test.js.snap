// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unzip positive 1`] = `
Array [
  "favicon.ico",
  "images/stop.svg",
  "images/u-turn.svg",
  "index.html",
  "static/css/2.a08beec1.chunk.css",
  "static/css/2.a08beec1.chunk.css.map",
  "static/css/main.af8ccdc5.chunk.css",
  "static/css/main.af8ccdc5.chunk.css.map",
  "static/js/main.975c9892.chunk.js",
  "static/js/runtime~main.9e52baa2.js",
]
`;
